cli/flox/src/commands/edit.rs

```diff
@@ -315,29 +315,52 @@ impl Edit {
         }
     }
 
-    /// Determines the editor to use for interactive editing
+    /// Determines the editor to use for interactive editing, based on the environment
     ///
-    /// If $EDITOR or $VISUAL is set, use that. Otherwise, try to find a known editor in $PATH.
+    /// If $VISUAL or $EDITOR is set, use that.
+    /// The editor cannot be an empty string or one that consists of fully Unicode whitespace.
+    /// Arguments can be passed and will be split on whitespace.
+    /// Otherwise, try to find a known editor in $PATH.
     /// The known editor selected is the first one found in $PATH from the following list:
     ///
     ///   vim, vi, nano, emacs.
-    fn determine_editor() -> Result<PathBuf> {
-        let editor = std::env::var("EDITOR").or(std::env::var("VISUAL")).ok();
+    fn determine_editor() -> Result<(PathBuf, Vec<String>)> {
+        Self::determine_editor_from_vars(
+            env::var("VISUAL").unwrap_or_default(),
+            env::var("EDITOR").unwrap_or_default(),
+            env::var("PATH").context("$PATH not set")?,
+        )
+    }
 
-        if let Some(editor) = editor {
-            return Ok(PathBuf::from(editor));
-        }
+    /// Determines the editor to use for interactive editing, based on passed values
+    fn determine_editor_from_vars(
+        visual_var: String,
+        editor_var: String,
+        path_var: String,
+    ) -> Result<(PathBuf, Vec<String>)> {
+        let var = if !visual_var.is_empty() {
+            visual_var
```

Comment:
> Fallback behavior is slightly weird/non-uniform when VISUAL is whitespace only -- do we care? If so, it will just require an additional whitespace parser for that case.

```suggestion
        let var = if !visual_var.trim().is_empty() {
```

yes i think we can do whitespace trimming which should make skipping somewhat more consistent

Reply:
this makes perfect sense -- updated test

Reply:
notably, we don't support whitespace only filenames.
this seems fine.